libs/journey-bundles/transactions/src/lib/transactions.bundle.ts

```diff
@@ -3,79 +3,60 @@ import {
   withConfig,
   withCommunicationService,
   withExtensions,
-  TRANSACTIONS_JOURNEY_CONFIG,
+  TransactionsJourneyExtensionsConfig,
 } from '@backbase/transactions-journey';
 import { JourneyCommunicationService } from '@backbase/shared/feature/communication';
-import { SHARED_JOURNEY_CONFIG } from '@backbase/shared/util/config';
-import { NgModule, Injectable, inject } from '@angular/core';
 import { TransactionItemAdditionalDetailsComponent } from './transaction-additional-details.component';
-import { RouterModule, Routes } from '@angular/router';
+import { Routes } from '@angular/router';
 import {
   TransactionsRouteTitleResolverService,
   TransactionsJourneyConfiguration,
 } from '@backbase/transactions-journey/internal/data-access';
+import { TRANSACTION_EXTENSIONS_CONFIG } from '@backbase/transactions-journey/internal/feature-transaction-view';
 
-// Create a service that will be used to configure the journey
-@Injectable()
-export class TransactionsConfigService {
-  readonly #slimMode =
-    inject(SHARED_JOURNEY_CONFIG, { optional: true })?.designSlimMode ?? false;
+// Extensions configuration
+const extensionsConfig: Partial<TransactionsJourneyExtensionsConfig> = {
+  transactionItemAdditionalDetails: TransactionItemAdditionalDetailsComponent,
+};
 
-  getJourneyConfig() {
-    return {
-      pageSize: 10,
-      slimMode: this.#slimMode,
-    };
-  }
-}
+// Journey configuration
+const journeyConfig = {
+  pageSize: 10,
+  slimMode: false,
+};
 
-// The actual routes that will be lazy-loaded
-const routes: Routes = transactionsJourney(
-  // Journey configuration - using a factory function that will be called at runtime
-  withConfig({
-    pageSize: 10,
-    slimMode: false, // Default value, will be updated by the module
-  }),
-  // Communication service configuration
+// Providers needed by the journey's internal services
+// Note: TransactionsJourneyConfiguration is an @Injectable class used by internal services
+// (TransactionsHttpService, TransactionsRouteTitleResolverService), which is different from
+// the TRANSACTIONS_JOURNEY_CONFIG InjectionToken provided by withConfig().
+const journeyProviders = [
+  {
+    provide: TransactionsJourneyConfiguration,
+    useValue: journeyConfig,
+  },
+  TransactionsRouteTitleResolverService,
+  {
+    provide: TRANSACTION_EXTENSIONS_CONFIG,
+    useValue: extensionsConfig,
+  },
```

Comment:
This provider for `TRANSACTION_EXTENSIONS_CONFIG` isn't needed as that's what `withExtensions` does below